sympy/algebras/quaternion.py

```diff
@@ -532,10 +534,16 @@ def to_euler(self, seq, angle_addition=True, avoid_square_root=False):
             a, b, c, d = a - c, b + d, c + a, d - b
 
         if avoid_square_root:
-            n2 = self.norm()**2 if symmetric else 2 * self.norm()**2
-            angles[1] = acos((a*a + b*b - c*c - d*d) / n2)
+            if symmetric:
+                n2 = self.norm()**2
+                angles[1] = acos(factor((a * a + b * b - c * c - d * d) / n2))
+            else:
+                n2 = 2 * self.norm()**2
+                angles[1] = asin(factor((c * c + d * d - a * a - b * b) / n2))
```

Comment:
I'm not sure if the `factor` has to be used than building the program with direct formula from symbolic `Quaternion(a, b, c, d)` value.

Reply:
What do you mean? That it's best to leave the `factor` out?

Reply:
Alternatively it can be a call to `simplify`, but I think it is usually best to avoid it and use something more direct instead, right?

Reply:
Anything about `Quaternion(expr1, expr2, expr3, expr4)` can be substituted from a single formula of 
`Quaternion(a, b, c, d).to_euler`

Reply:
So I think why not it's better to give result for arbitrary formula to be compatible with substitution. it can be an options to pick arbitrary `simplify, factor` to break it further down